### PR TITLE
feat: storing and loading data with json

### PR DESCRIPTION
**Pull request by AgelxNash**
**will loose compatibility with the previous save**

### CODE REVIEW

File: src/RequestLogEntry.php [hunk 1]
```diff
@@ -32,9 +32,9 @@ class RequestLogEntry extends Model
         'path' => 'string',
         'method' => 'string',
         'agent' => 'string',
-        'get' => 'string',
-        'post' => 'string',
-        'cookies' => 'string',
+        'get' => 'json',
```
Comment (amenk) [inline src/RequestLogEntry.php:35]: since which Laravel Version is casting to json working? I think we should add the requirement in composer.json ...

**Comment by AgelxNash** [comment]
Any news?

**Comment by amenk** [comment]
@AgelxNash  Sorry, I forgot to finish the review, so my comments were not yet visible

File: src/RequestLogEntry.php (hunk 1 above)
Comment (AgelxNash) [reply src/RequestLogEntry.php:35]: 5.0 https://github.com/illuminate/database/blob/5.0/Eloquent/Model.php#L2755

**Comment by amenk** [comment]
@nachtsheim @DanieliMi @UB3RL33T Do you think we should switch storage format to JSON?

File: src/LogRequest.php [hunk 2]
```diff
@@ -24,21 +24,29 @@ public function handle($request, Closure $next)
...
+            RequestLogEntry::create($this->getData());
         }
 
         return $next($request);
     }
 
+    /**
+     * @return array
+     */
+    protected function getData() : array
```
Comment (DanieliMi) [inline src/LogRequest.php:36]: It seems the composer.json is missing some important information. Due to the return types the package will only work with php < 7. Can you please add the php requirements to composer.json?

Comment (AgelxNash) [reply src/LogRequest.php:36]: https://github.com/doctrine/dbal/blob/v2.9.0/composer.json#L24-L27

Comment (DanieliMi) [reply src/LogRequest.php:36]: Fair enough didn't have that in mind.

**Pull request merged**